lib/index.js

```diff
@@ -1,111 +0,0 @@
-(function (global, factory) {
-  if (typeof define === "function" && define.amd) {
-    define(['module', 'exports', 'camel-case'], factory);
-  } else if (typeof exports !== "undefined") {
-    factory(module, exports, require('camel-case'));
-  } else {
-    var mod = {
-      exports: {}
-    };
-    factory(mod, mod.exports, global.camelCase);
-    global.index = mod.exports;
-  }
-})(this, function (module, exports, _camelCase) {
-  'use strict';
-
-  Object.defineProperty(exports, "__esModule", {
-    value: true
-  });
-
-  exports.default = function (CustomElement) {
-    var opts = arguments.length <= 1 || arguments[1] === undefined ? {} : arguments[1];
-
-    var React = opts.React || window.React;
-    var ReactDOM = opts.ReactDOM || window.ReactDOM;
-    var container = opts.container || 'div';
-    var defaultProps = getDefaultProps(CustomElement.properties);
-
-    var ReactClass = React.createClass({
-      getDefaultProps: function getDefaultProps() {
-        return defaultProps;
-      },
-      render: function render() {
-        return React.createElement(container);
-      },
-      renderChildren: function renderChildren(props) {
-        // Render the component in the new tree
-        ReactDOM.render(React.createElement(container, null, props.children), this._realCustomElement);
-
-        // Passes on all non-react-special props to the custom element.
-        setCustomElementProps(this._realCustomElement, props);
-      },
-      componentDidMount: function componentDidMount() {
-        var props = this.props;
-
-        // The real custom element is the component that we want to contain the
-        // new render tree as its content.
-        this._realCustomElement = CustomElement(makeCustomElementProps(props));
-
-        // Listen for custom element changes and cancel them to make a stateless component
-        this._realCustomElement.addEventListener('property-change', function (e) {
-          if (props.handlePropertyChange) {
```

Comment:
I ended up keeping it and writing tests for it. In doing so, I realised all it's really doing is automating the binding to the `prop-change` event and calling `e.preventDefault()`. Now that we support custom events, all you have to do is simply bind to the event and call `e.preventDefault()` yourself. This prevents us from having to maintain and test this behaviour, and we'd also want to allow customisation of the event name and prop handler name. Now all you need to do is:

```js
<Component onPropChange={e => e.preventDefault()} />
```

...if `propchange` was emitted when a property changes.